src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -138,3 +156,42 @@ AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt)
 		}
 	}
 }
+
+
+/*
+ * AppendAlterRoleStmt generates the string representation of the
+ * AlterRoleStmt and appends it to the buffer.
+ */
+static void
+AppendAlterRoleSetStmt(StringInfo buf, AlterRoleSetStmt *stmt)
+{
+	RoleSpec *role = stmt->role;
+	const char *roleSpecStr = NULL;
+
+	if (role == NULL)
+	{
+		/*
+		 * If all roles are be affected, role field is left blank in an
+		 * AlterRoleSetStmt.
+		 */
+		roleSpecStr = "ALL";
+	}
+	else
+	{
+		/*
+		 * If the role_specification used is CURRENT_USER or SESSION_USER,
+		 * it will be converted to thats roles role name.
+		 */
+		roleSpecStr = RoleSpecString(role, true);
+	}
+
+	appendStringInfo(buf, "ALTER ROLE %s", roleSpecStr);
+
+	if (stmt->database != NULL)
+	{
+		appendStringInfo(buf, " IN DATABASE %s", stmt->database);
```

Comment:
can you double check if we need to quote this

Reply:
We definitely need to quote this.